governance/pyth_staking_sdk/src/utils/bn.ts

```diff
@@ -0,0 +1,45 @@
+import { BN } from "@coral-xyz/anchor";
+import type { ConvertBigIntToBN, ConvertBNToBigInt } from "../types";
+
+export const convertBNToBigInt = <T>(obj: T): ConvertBNToBigInt<T> => {
+  if (obj instanceof BN) {
+    return BigInt(obj.toString()) as any;
+  }
+  if (Array.isArray(obj)) {
+    return obj.map(convertBNToBigInt) as any;
+  }
+  if (typeof obj !== "object" || obj === null || obj.constructor !== Object) {
+    return obj as any;
```

Comment:
what's the point of typecasting to `any`?  `any` is an absolutely toxic type, I would be really concerned about any code that requires it to work.

Would it be more appropriate to type this as something along these lines:

```ts
const ConvertibleToBigInt = BN | BN[] | Record<string, ConvertableToBigInt>;

const convertBNToBigInt = <T extends ConvertableToBigInt>(obj: T): ConvertBNToBigInt<ConvertableToBigInt> => {
    ...
}
```

?

Reply:
I 100% agree with you. I tried for hours trying to avoid any with no luck. Your suggestion doesn't work because our object can literally have anything and anything is ConvertibleToBigInt. For example, `a: {b: BN, c: Pubkey}` here Pubkey should be `ConvertibleToBigInt` because of `Record<string, ConvertableToBigInt>`.